$OpenBSD: patch-src_math-equation_c,v 1.1 2011/06/01 15:07:48 ajacoutot Exp $

Fix stackoverflow as the lexer wouldn't be able to fit on the stack.
https://bugzilla.gnome.org/show_bug.cgi?id=650174

--- src/math-equation.c.orig	Tue May 24 05:20:42 2011
+++ src/math-equation.c	Tue May 31 09:37:25 2011
@@ -1305,6 +1305,7 @@ void
 math_equation_solve(MathEquation *equation)
 {
     GError *error = NULL;
+    gulong stacksize = 0;
 
     g_return_if_fail(equation != NULL);
 
@@ -1326,7 +1327,14 @@ math_equation_solve(MathEquation *equation)
 
     math_equation_set_number_mode(equation, NORMAL);
 
-    g_thread_create(math_equation_solve_real, equation, false, &error);
+    if (GLIB_SIZEOF_LONG == 8) {
+           stacksize = 0x400000;
+    } else {
+           stacksize = 0x200000;
+    }
+
+    g_thread_create_full(math_equation_solve_real, equation, stacksize, false,
+        false, G_THREAD_PRIORITY_NORMAL, &error);
 
     if (error)
         g_warning("Error spawning thread for calculations: %s\n", error->message);
